### PR TITLE
Try different classes as holder for the mocking bridge fields

### PR DESCRIPTION
**Pull request by Vampire**
If you load JMockit dynamically after something already loaded the class `java.lang.NegativeArraySizeException`, the mocking bridge fields will not get generated and you get an error e. g. about missing field `$MMB` if you run your tests.

Issue #298 is one example of this happening, because Hazelcast got initialized before JMockit is initialized and Hazelcast loaded this class already.

Another example would be another application loaded in the same application server before that used this class already.

This PR tries to instrument `java.lang.NegativeArraySizeException`, `java.lang.annotation.IncompleteAnnotationException`, `java.lang.OutOfMemoryError` and `java.lang.StackOverflowError` until it founds one that the instrumentation worked with and throws an error otherwise. This list can be extended as you like with additional classes to be tried.

### CODE REVIEW

**Comment by rliesenfeld** [comment]
Needs repro test.

**Comment by Vampire** [comment]
Test added

File: main/src/mockit/internal/startup/MockingBridgeFields.java [hunk 1]
```diff
@@ -12,38 +12,93 @@
 import mockit.internal.*;
 import mockit.internal.expectations.mocking.*;
 import mockit.internal.mockups.*;
+import mockit.internal.util.*;
```
Comment (marcdonovan) [inline main/src/mockit/internal/startup/MockingBridgeFields.java]: http://javadude.com/articles/importondemandisevil.html

**Comment by rliesenfeld** [comment]
Implemented without having to expect any known JRE classes to be still unloaded. Thanks for the opening the issue.

File: main/src/mockit/internal/startup/MockingBridgeFields.java (hunk 1 above)
Comment (Vampire) [inline main/src/mockit/internal/startup/MockingBridgeFields.java]: Actually I also think star-imports are bad, as I want to see from the imports instantly where a referenced class is coming from, even if I'm not within an IDE. But if I contribute to foreign codebases, I usually adjust to the code-style I find and JMockit uses star-import everywhere.

**Comment by Vampire** [comment]
Ah, nice solution to just use the first class that gets loaded.
One thought so: Now it is not really safe that this is a JRE-class that comes from the system class-loader, is it?
Isn't this essential for the principle to work?
Maybe you should check additionally that it is a `java.` or `javax.` class and that it comes fromt he system classloader, or something like that?

PS: Are you aware that you don't push any tags to GitHub? It would be nice if you could push your tags, so that it is clear from which revision a version was built exactly.

**Comment by rliesenfeld** [comment]
It already checks the class is a JRE class ("loader == null").

I don't see a concrete advantage to use tags, so I keep it simple.

**Comment by Vampire** [comment]
> It already checks the class is a JRE class ("loader == null").

Ah, I see, great. :-)

> I don't see a concrete advantage to use tags, so I keep it simple.

Well, it makes it easier to find which revision a version was built from.
E. g. i built an own version to play with the PRs I made, but for our project I wanted to base them off an officially release version. So I had to find out first which commit the 1.24 was built from.

Of course I could also have used the sources artifact, but then I don't have the repository information in there and so on. But ok, I thought you just missed to push them as they are not pushed by default in Git. But if you just don't use tags and don't want to, there is of course not much that can be done here. ;-)

**Pull request closed without merging**